src/components/Router/HistoryRouter.tsx

```diff
@@ -220,10 +221,8 @@ export default class HistoryRouter extends Component<
 
   getDocumentType = (): DocumentTypes | undefined => {
     const { documentType, steps } = this.props
-    const documentStep = steps?.find(
-      (step) => step.type === 'document'
-    ) as StepConfigDocument
 
+    const documentStep = buildStepFinder(steps)('document')
```

Comment:
Minor: I think it's clearer/readable to break this line down into 2 like elsewhere in codebase where this is being used.
```suggestion
    const findStep = buildStepFinder(steps)
    const documentStep = findStep('document')
```
or maybe even make it just 1 function call if possible rather than having to always do this 2-function-call combo each time we want to find a step
```suggestion
    const documentStep = buildStepFinder(steps, 'document')
```
Though appreciate I might've missed a detail in the bigger picture/scope of this implementation here & there's a reason it is the way it is.

Reply:
This feature is called [currying](https://medium.com/javascript-scene/curry-and-function-composition-2c208d774983), which is a good practice to reuse complex utilities. I'll keep the signature and break the usage for better readability.

Reply:
Yeah, I was initially confused about why that was written that way then realised what you're doing there. More hesitant about changing the function signature to the 2nd suggestion but thought I'd suggest it anyway.